### PR TITLE
metrics: use uint64 for Counter and Gauge types

### PR DESCRIPTION
**Pull request by cce**
## Summary

Our counters and gauges support float64, but we never use anything but uint64, so this switches them to be uint64 metrics.

## Test Plan

Updated tests in the metrics package.

### CODE REVIEW

File: util/metrics/gauge.go [hunk 1]
```diff
@@ -60,14 +60,14 @@ func (gauge *Gauge) Deregister(reg *Registry) {
 }
 
 // Add increases gauge by x
-func (gauge *Gauge) Add(x float64) {
+func (gauge *Gauge) Add(x uint64) {
 	gauge.Lock()
```
Comment (brianolson) [inline util/metrics/gauge.go:64]: go ahead and use `sync/atomic`?

File: util/metrics/gauge.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ type Gauge struct {
 	deadlock.Mutex
 	name        string
 	description string
-	value       float64
+	value       uint64
```
Comment (brianolson) [inline util/metrics/gauge.go:31]: move to first member of struct for better `sync/atomic` safety?

File: util/metrics/counter.go [hunk 3]
```diff
@@ -66,13 +66,12 @@ func (counter *Counter) Inc(labels map[string]string) {
...
-		counter.Add(1.0, labels)
+		counter.addLabels(1.0, labels)
 	}
 }
 
-// Add increases counter by x
-// For adding an integer, see AddUint64(x)
-func (counter *Counter) Add(x float64, labels map[string]string) {
+// addLabels increases counter by x
+func (counter *Counter) addLabels(x uint64, labels map[string]string) {
```
Comment (brianolson) [inline util/metrics/counter.go:74]: I like that this was changed and nothing was lost because everything was already using AddUint64

**Review by brianolson — COMMENTED** [review]
99% love it. about time. but we should go ahead and use `sync/atomic` in this change

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**